packages/frontend-core/src/components/Updating.svelte

```diff
@@ -0,0 +1,61 @@
+<script>
+  export let isMigrationDone
+  export let onMigrationDone
+  export let timeoutMs = 180000 // 3 minutes
+
+  const loadTime = Date.now()
+  let timedOut = false
+
+  async function checkMigrationsFinished() {
+    setTimeout(async () => {
+      const isMigrated = await isMigrationDone()
+      if (!isMigrated) {
+        if (loadTime + timeoutMs > Date.now()) {
+          return checkMigrationsFinished()
+        }
+
+        return migrationTimeout()
+      }
+
+      onMigrationDone()
+    }, 1000)
+  }
+
+  checkMigrationsFinished()
+
+  function migrationTimeout() {
+    timedOut = true
+  }
+
+  $: text = !timedOut ? "System update" : "Something went wrong!"
+  $: subtext = !timedOut
+    ? "Please wait and we will be back in a second!"
+    : "An error occurred, please try again later"
```

Comment:
Is there something more helpful we can say here? Ask them to email support and give them some logs to show them, maybe? I can see users being frustrated here if it happens repeatedly and they're locked out of their app.